### PR TITLE
Add assignee endpoint for view and update requests

### PR DESCRIPTION
**Pull request by igarcez**
This MR adds a new endpoint for an assignee to view its requests at `/api/v1/me/assigned-requests/`and `/api/v1/me/assigned-requests/[id]`

### CODE REVIEW

File: management/views.py [hunk 1]
```diff
@@ -46,3 +46,19 @@ def destroy(self, request, *args, **kwargs):
...
+    permission_classes = [IsAuthenticated]
+    serializer_class = RequestSerializer
+
+    def get_queryset(self):
+        return Request.objects.filter(assignee__user=self.request.user)
+
+
+class MeAssignedRequestViewUpdateAPIV1(generics.RetrieveUpdateAPIView):
+    permission_classes = [IsAuthenticated, IsAssigneeOfRequest]
+    serializer_class = RequestSerializer
```
Comment (tonythomas01) [inline management/views.py:61]: yeah, lets disable PUT here, by using this: 
`http_method_names = ["get", "patch", "options"]`

File: management/urls_v1.py [hunk 2]
```diff
@@ -4,10 +4,14 @@
...
+    MeAssignedRequestsAPIV1,
+    MeAssignedRequestViewUpdateAPIV1
 )
 
 urlpatterns = [
     path("", MeRetrieveUpdateAPIViewV1.as_view(), name="me_retrieve_update"),
     path("requests/", ListCreateRequestsAPIV1.as_view(), name="create_list_requests"),
     path("requests/<int:pk>/", MeRequestDetailAPIV1.as_view(), name="detail_request"),
+    path("assigned-requests/", MeAssignedRequestsAPIV1.as_view()),
+    path("assigned-requests/<int:pk>", MeAssignedRequestViewUpdateAPIV1.as_view()),
```
Comment (tonythomas01) [inline management/urls_v1.py]: I usually put a `/` at the end.

File: management/views.py (hunk 1 above)
Comment (tonythomas01) [reply management/views.py:61]: We need to also see how we can limit the write functionality of an assignee to be limited to only change `status`. There should be some serializer magic which we can use. Or, just mark everything else as read_only HAHA

**Review by tonythomas01 — COMMENTED** [review]
Some minor comments. Good shit.

File: management/views.py (hunk 1 above)
Comment (igarcez) [reply management/views.py:61]: both points are addressed now, can you please check again @tonythomas01

File: management/views.py [hunk 3]
```diff
@@ -40,9 +40,32 @@ class MeRequestDetailAPIV1(generics.RetrieveUpdateDestroyAPIView):
...
+class MeAssignedRequestsAPIV1(generics.ListAPIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = RequestSerializer
+
+    def get_queryset(self):
+        return Request.objects.filter(assignee__user=self.request.user)
+
+
+class MeAssignedRequestViewUpdateAPIV1(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated, IsAssigneeOfRequest]
```
Comment (tonythomas01) [inline management/views.py:60]: is there a reason we inherit from `viewsets.ModelViewSet` and not `generics.RetrieveUpdateAPIView` ?

File: management/urls_v1.py (hunk 2 above)
Comment (igarcez) [reply management/urls_v1.py]: Forgot about this one, sorry, pushed the change now

**Review by tonythomas01 — APPROVED** [review]
LGTM. Good stuff and ✔️

**Pull request merged**